src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -359,6 +367,73 @@ def get_package_size(self, output):
             return Constants.UNKNOWN_PACKAGE_SIZE
     # endregion
 
+    # region auto OS updates
+    def disable_auto_os_update(self):
+        """ Disables auto OS updates on the machine only if they are enabled and logs the default settings the machine comes with """
+        try:
+            self.composite_logger.log_debug("Disabling auto OS updates if they are enabled")
+            self.backup_image_default_patch_mode()
+            self.update_image_default_patch_mode(self.update_package_list, "0")
+            self.update_image_default_patch_mode(self.unattended_upgrade, "0")
+            self.composite_logger.log("Successfully disabled auto OS updates")
+        except Exception as error:
+            self.composite_logger.log_error("Could not disable auto OS updates. [Error={0}]".format(repr(error)))
+
+    def backup_image_default_patch_mode(self):
+        """ Records the default system settings for auto OS updates within patch extension artifacts for future reference.
+        We only log the default patch_mode a VM comes with, any subsequent updates will not be recorded"""
+        try:
+            if not self.image_default_patch_mode_backup_exists():
```

Comment:
Include an else block and log that a backup already exists.

Reply:
Logging that within image_default_patch_mode_backup_exists(), line 345 in PackageManager.py